Xplat/src/main/java/vazkii/botania/data/LooniumStructureLootProvider.java

```
@@ -179,7 +179,7 @@ public CompletableFuture<?> run(@NotNull CachedOutput cache) {
 		for (var e : tables.entrySet()) {
 			Path path = pathProvider.json(e.getKey());
 			LootTable.Builder builder = e.getValue();
-			LootTable lootTable = builder.setParamSet(LootContextParamSets.EMPTY).build();
+			LootTable lootTable = builder.setParamSet(LootContextParamSets.ALL_PARAMS).build();
```

Shouldn't this ideally be changed to the context parameters we actually supply/use?

---

IIRC there's validation happening when the loot table is written, loaded, or evaluated. And while you don't need to set any parameters, having them not defined in the first place can lead to warnings or errors if the type of the referenced loot requires them.
The Loonium loot is generated in an empty context (it could probably be a different context that provides the flower location and mob reference somehow, but that could influence the type of items you get, e.g. due to cooking food drops if the mob was on fire, or whether it was a player kill with or without looting), but there are references to e.g. entity, chest, or archaeology type loot tables, which may have different parameter sets.

Long story short: This commit changed it from the actually supplied set (empty) to the widest available set (all params) because something didn't work as expected.

---

Could maybe be solved with a special loot function that makes the table ignore errors/warnings for missing parameters in a smooth way, that can be applied if the source loot table expects some parameters. But specifying all_params is probably fine for now